engine/execution/messages.go

```diff
@@ -11,38 +11,118 @@ import (
 // TODO(patrick): rm unaccessed fields
 type ComputationResult struct {
 	*entity.ExecutableBlock
+	*BlockExecutionResults
+	*BlockAttestationResults
+
+	*flow.ExecutionReceipt
+}
+
+func NewEmptyComputationResult(
+	block *entity.ExecutableBlock,
+) *ComputationResult {
+	return &ComputationResult{
+		ExecutableBlock:         block,
+		BlockExecutionResults:   NewEmptyBlockExecutionResults(block),
+		BlockAttestationResults: NewEmptyBlockAttestationResults(block),
+	}
+}
+
+func (cr *ComputationResult) CollectionExecutionResult(colIndex int) *CollectionExecutionResult {
+	if colIndex < 0 && colIndex > len(cr.StateSnapshots) {
+		return nil
```

Comment:
Better to return `nil, fmt.Errorf("index %v out of range [0, %v]")`.

This could remind the caller to handle the `nil` case